Cython/Compiler/Code.py

```diff
@@ -1584,6 +1631,34 @@ def finalize_main_c_code(self):
         code.putln(util.format_code(util.impl))
         code.putln("")
 
+        shared_module_loaded = self.module_node.scope.context.shared_utility_qualified_name
+        shared_module_generated = self.module_node.scope.context.options.shared_c_file_path
```

Comment:
Ah, seeing this combination now helped me understand that the variable names intend to distinguish the two cases of using or being the shared module, right? That was totally not clear to me before. Then I suggest:
```suggestion
        using_shared_utility_module = bool(self.module_node.scope.context.shared_utility_qualified_name)
        in_shared_utility_module = bool(self.module_node.scope.context.options.shared_c_file_path)
```

But please move this entire code section out of `Code.py`. It seems a better fit as separate methods in `ModuleNode` for the "in" and "using" case, called before `finalize_main_c_code()`, rather than being a part of this method here. It doesn't have to do anything with the finalisation, it's really something that is part of the module's functionality.